pkg/diagnose/deployments.go

```diff
@@ -68,10 +70,20 @@ func checkOverlappingCIDRs(clusterInfo *cluster.Info, status reporter.Interface)
 
 	defer status.End()
 
+	brokerRestConfig, brokerNamespace, err := restconfig.ForBroker(clusterInfo.Submariner, nil)
+	if err != nil {
+		return status.Error(err, "Error getting the Broker's REST config")
+	}
+
+	clientProducer, err := client.NewProducerFromRestConfig(brokerRestConfig)
+	if err != nil {
+		return status.Error(err, "Error creating broker client Producer")
+	}
+
 	endpointList := &submarinerv1.EndpointList{}
 
-	err := clusterInfo.ClientProducer.ForGeneral().List(context.TODO(), endpointList,
-		controllerClient.InNamespace(clusterInfo.Submariner.Namespace))
+	err = clientProducer.ForGeneral().List(context.TODO(), endpointList,
+		controllerClient.InNamespace(brokerNamespace))
 	if err != nil {
 		return status.Error(err, "Error listing the Submariner endpoints")
```

Comment:
Overall code LGTM, I have one comment/Q,
What if the broker is down? maybe we can update the error message?

Reply:
If there is an error while connecting to the Broker, the `err` message should have the appropriate details.

Reply:
Let me add some note saying that its while reading from the Broker.